src/SFA.DAS.Charities.Import/Functions/ImportCharityCommissionData/Activities/ImportCharityCommissionDataActivity.cs

```diff
@@ -0,0 +1,65 @@
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.Charities.Import.Functions.ImportCharityCommissionData.Activities
+{
+    public class ImportCharityCommissionDataActivity
+    {
+        private readonly string downloadUrl;
+        private readonly HttpClient _httpClient;
+
+        public ImportCharityCommissionDataActivity(IConfiguration configuration, HttpClient httpClient)
+        {
+            downloadUrl = configuration["CharityFilesDownloadUrl"];
+            _httpClient = httpClient;
+        }
+
+        [FunctionName(nameof(ImportCharityCommissionDataActivity))]
+        public async Task Run(
+            [ActivityTrigger] string fileName, 
+            [Blob("charity-files/{fileName}", FileAccess.Write)] Stream file,
+            ILogger log)
+        {
+            var charityFileDownloadUri = new Uri(Path.Combine(downloadUrl, fileName));
```

Comment:
Normally you set the base address during Dependency injection setup and have it as either an named instance or specified Interface

Reply:
Fixed